src/JSInterop/Microsoft.JSInterop/src/IJSDataReference.cs

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.JSInterop
+{
+    /// <summary>
+    /// Represents a reference to JavaScript data to be consumed through a stream.
+    /// </summary>
+    public interface IJSDataReference : IAsyncDisposable
+    {
+        /// <summary>
+        /// Length of the stream provided by JavaScript.
+        /// </summary>
+        public long Length { get; }
+
+        /// <summary>
+        /// Initiatializes a <see cref="Stream"/> with the <see cref="JSRuntime"/> for the current data reference.
+        /// </summary>
+        /// <param name="maxAllowedSize">Maximum number of bytes permitted to be read from JavaScript.</param>
+        /// <param name="maxBufferSize">Amount of bytes to buffer before flushing.</param>
```

Comment:
I might be wrong about what this means, but should this be

```suggestion
        /// <param name="maxBufferSize">The maximum number of bytes that are allowed to be buffered.</param>
```